low-level/actuators/impl/generic/node_hw.py

```diff
@@ -225,29 +223,24 @@ def _get_sensor_properties(self, sensor_name):
          Deassertions Enabled  : unc+ ucr+
         """
         try:
-            sensor_get_response, return_code = self._executor._run_ipmitool_subcommand("sensor get '{0}'".format(sensor_name))
+            cmd = f"sensor get '{sensor_name}'"
+            sensor_get_response, err, return_code = \
+                self._executor._run_ipmitool_subcommand(cmd)
             if return_code == 0:
                 return self._response_to_dict(sensor_get_response)
             else:
-                msg = "sensor get '{0}' : command failed with error {1}".format(sensor_name, sensor_get_response)
+                msg = (f"sensor get '{sensor_name}' :"
+                        " command failed with error {err}")
                 logger.warn(msg)
                 return self._errorstr_to_dict(sensor_get_response)
         except Exception as err:
             logger.error("Exception occurred in _get_sensor_properties for cmd - sensor get '{0}': {1}".format(sensor_name, err))
 
-    def _get_str_response(self, data):
-        # check if data is tuple, convert to string
-        if isinstance(data, tuple):
-            data_str = ''.join([item.decode("utf8") for item in data])
-        else:
-            data_str = data.decode("utf-8")
-        return data_str
 
     def _errorstr_to_dict(self, data):
         error_resp = {'sensor_error': None}
         try:
-            data_str = self._get_str_response(data)
-            for line in data_str.split("\n"):
+            for line in data.split("\n"):
```

Comment:
Why did we remove the call to `get_str_response`? can we assume we won't get a tuple for `data`? Also why are spliting by `\n` which we didn't do earlier? 
Okay. I could see you have covered it as part of _run_ipmitool_subcommand. we need more testing for this function.

Reply:
For why I removed `get_str_sesponse` please refer previous comment history, we were splitting it earlier also.
Unit tests can be further added for the function `_run_ipmitool_subcommand()` currently I have added only two,
these unit tests we will be executing on vm only I guess and on vm we can test ipmi over kcs/lan  and via ipmisimtool.
out of these over lan impl is remaining.